data/transactions/logic/assembler.go

```diff
@@ -1125,6 +1132,51 @@ func typeTxField(pgm ProgramKnowledge, args []string) (StackTypes, StackTypes) {
 	return StackTypes{fs.ftype}, nil
 }
 
+func typeStore(pgm *ProgramKnowledge, args []string) (StackTypes, StackTypes) {
+	n, errored := getUintImm(args)
+	if errored {
+		return nil, nil
+	}
+	scratchIndex := int(n)
+	top := len(pgm.stack) - 1
+	if top >= 0 {
+		pgm.scratchSpace[scratchIndex] = pgm.stack[top]
+	}
+	return nil, nil
+}
+
+func typeStores(pgm *ProgramKnowledge, args []string) (StackTypes, StackTypes) {
+	top := len(pgm.stack) - 1
+	if top < 0 {
+		return nil, nil
+	}
+	for i := range pgm.scratchSpace {
+		if pgm.scratchSpace[i] != pgm.stack[top] {
+			pgm.scratchSpace[i] = StackAny
```

Comment:
add a `break` here?

Reply:
I'm not sure what you mean, but I also definitely should have commented this section of the code, so I'll put my thinking here and then also go and comment this section.

The thinking here was that if they do a stores, we can't know which scratch index they're putting top into, but we can at least keep the ones that are the same type as top while the rest get set to StackAny

Reply:
Haha, no worries, I think that just says a lot about my lack of commenting